crates/typst/src/geom/angle.rs

```diff
@@ -85,6 +109,27 @@ impl Angle {
     pub fn to_deg(self) -> f64 {
         self.to_unit(AngleUnit::Deg)
     }
+
+    /// Get the quadrant of the Cartesian plane that this angle lies in.
+    /// The angle is automatically normalized to the range `0deg..=360deg`.
+    /// The quadrants are defined as follows:
+    /// - First: `0deg..=90deg` (top-right)
+    /// - Second: `90deg..=180deg` (top-left)
+    /// - Third: `180deg..=270deg` (bottom-left)
+    /// - Fourth: `270deg..=360deg` (bottom-right)
+    #[func]
+    pub fn quadrant(self) -> Quadrant {
+        let rad = self.to_rad().rem_euclid(TAU);
+        if (0.0..=FRAC_PI_2).contains(&rad) {
+            Quadrant::First
+        } else if (FRAC_PI_2..=PI).contains(&rad) {
+            Quadrant::Second
+        } else if (PI..=(3.0 * FRAC_PI_2)).contains(&rad) {
+            Quadrant::Third
+        } else {
+            Quadrant::Fourth
+        }
```

Comment:
would this logic perhaps be simpler in `to_deg()`?

Reply:
I mean same same, I just write it in a less convoluted way ;)